cosmoz-default-tree.html

```diff
@@ -14,143 +27,249 @@
+		var results = [];
 
-		if (propertyName === 'pathLocator') {
-			return this._getPathByPathLocator(propertyValue);
-		}
+		// Defaults
+		nodes = nodes || this._roots;
 
-		var node = this._searchAllRoots(propertyName, propertyValue);
-		if (node !== undefined) {
-			return this._getPathByPathLocator(node.pathLocator);
-		}
+		nodes.some(function (node) {
+			results = results.concat(this.search(node, propertyValue, options));
+			return options.firstHitOnly && results.length > 0;
+		}, this);
+
+		return results;
 	};
 
-	Cosmoz.DefaultTree.prototype.getChildren = function (node) {
-		if (node === undefined) {
-			return;
+	/**
+	 * Returns the node of a given path.
+	 * @return {object}
+	 * @param {string} pathLocator (The string which describes the path. e.g. "1.2.9")
+	 * @param {object} nodeObj [this._treeData] (The object the search should be based on.)
+	 * @param {string} pathLocatorSeparator [this.pathLocatorSeparator] (The string which separates the path. e.g ".")
+	*/
+	Cosmoz.DefaultTree.prototype.getNodeByPathLocator = function (pathLocator, nodeObj, pathLocatorSeparator) {
+		if (!pathLocator) {
+			return this._roots;
 		}
-		return _objectValues(node.children);
+
+		// Defaults
+		nodeObj = nodeObj || this._treeData;
+		pathLocatorSeparator = pathLocatorSeparator || this.pathLocatorSeparator;
+
+		var	pathNodes = this.getPathNodes(pathLocator, nodeObj, pathLocatorSeparator);
+		return pathNodes && pathNodes.pop();
 	};
 
-	Cosmoz.DefaultTree.prototype.getProperty = function (node, propertyName) {
-		if (node === undefined || propertyName === undefined) {
-			return;
+	/**
+	 * Returns the nodes on a given path.
+	 * A valid path 1.2.3 should return the items [1, 2, 3]
+	 * - path 1.2.3.3 should return [1, 2, 3, undefined]
+	 * - path 0.1.2.3 should return [1, 2, 3]
+	 * - path 0.1.5.3 should return [1, undefined, undefined]
+	 * @return {array}
+	 * @param {string} pathLocator (The string which describes the path. e.g. "1.2.9")
+	 * @param {object} nodeObj [this._treeData] (The object the search should be based on.)
+	 * @param {string} pathLocatorSeparator [this.pathLocatorSeparator] (The string which separates the path.)
+	*/
+	Cosmoz.DefaultTree.prototype.getPathNodes = function (pathLocator, nodeObj, pathLocatorSeparator) {
+		if (!pathLocator) {
+			return this._roots;
 		}
 
-		return node[propertyName];
-	};
+		// Defaults
+		pathLocatorSeparator = pathLocatorSeparator || this.pathLocatorSeparator;
 
-	Cosmoz.DefaultTree.prototype._searchAllRoots = function (propertyName, propertyValue) {
-		var i,
+		var path = pathLocator.split(pathLocatorSeparator),
+			pathSegment = nodeObj || this._treeData,
+			nodes = [],
 			node;
 
-		for (i = 0; i < this._roots.length; i+=1) {
-			node = this.search(this._roots[i], propertyName, propertyValue);
-			if (node !== undefined) {
-				return node;
+		// Get the nodes on the path
+		path.forEach(function (nodeKey) {
```

Comment:
Right now we perform the check against the object tree. So whichever "root object" has the first matching key, gets taken.

```
{
    "2": {
        "id": "167d1485-7d4f-4c7d-86cd-a4fb00f31245",
        "name": "Node2",
        "pathLocator": "1.2",
        "children": {}
    },

    "301": {
        "id": "3a7654f1-e3e6-49c7-b6a8-a4fb00f31245",
        "name": "Node301",
        "pathLocator": "1.2.3.301",
        "children": {
            "401": {
                "id": "865065da-f44c-472e-a8df-a4fb00f3124b",
                "name": "Node401",
                "pathLocator": "1.2.3.301.401",
                "children": {
                }

            }
        }
    }
}
```

- 1.2.3.301 returns [2, undefined, 301]
- 1.2 returns [2]
- 2.301 returns [2, 301]

What should be changed regarding the behavior?